CDDL files

GeorgeFlerovsky opened the pull request:
Here's what I have for our CIP's CDDL, right now.

TODO:

- [ ] modify README.md to discuss some of the specifics of the CDDL
- [ ] make some remaining decisions about the CDDL

GeorgeFlerovsky committed:
Demote top level headers to H2 + add H1 title

https://github.com/cardano-foundation/CIPs/pull/430#issuecomment-1386607038

GeorgeFlerovsky committed:
Fix internal section links

https://github.com/cardano-foundation/CIPs/pull/430#discussion_r1084572077

GeorgeFlerovsky committed:
Fix link/code typos

GeorgeFlerovsky committed:
Add CDDL draft file

nicolasayotte commented:
We agreed to add a bit of precision on the asset label types. Something like
```
bytes_32 = bytes .size (0..32)
label_utf8_32 = text .size (0..32) ; utf-8 encoded
label_hex_64 = text .size (0..64) ; hex encoded bytes_32
```
then merge.

nicolasayotte commented:
This will allow the CSV format to support both types of labels, probably used a `version` tag to tell the user what to expect

GeorgeFlerovsky committed:
Isolate CDDL for text-based (policy_id,asset_name)

This corresponds to the CIP-25's version_1.cddl.

GeorgeFlerovsky committed:
Add support for bytestring (policy_id, asset_name)

This corresponds to CIP-25's version_2.cddl.
Note that regex and tabular updates are not supported under this
version 2 mode.

GeorgeFlerovsky commented:
> This will allow the CSV format to support both types of labels, probably used a `version` tag to tell the user what to expect

I split the CDDL file into two files (`version_1.cddl` and `version_2.cddl`). In version two, the `action` object sets `version` to `2`.

nicolasayotte commented on CIP-????/cddl/version_1.cddl:
```diff
@@ -0,0 +1,153 @@
+; Fixed-size text and bytes
+label_32 = bytes_32 / text_32
+bytes_32 = bytes .size (0..32)
+text_32 = bytes .size (0..32)
```
text .size (0..32) ?

nicolasayotte commented on CIP-????/cddl/version_2.cddl:
```diff
@@ -0,0 +1,85 @@
+; Fixed-size text and bytes
+label_32 = bytes_32 / text_32
+bytes_32 = bytes .size (0..32)
+text_32 = bytes .size (0..32)
```
text again

nicolasayotte committed:
Update version_1.cddl

typo in text_32 definition

nicolasayotte committed:
Update version_2.cddl

typo in text_32 definition
